src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -944,34 +944,33 @@ await using (var gzipStream = await RavenServerBackupUtils.GetDecompressionStrea
                 SkipRevisionCreation = true
             };
 
-            var lastPath = GetSmugglerBackupPath(smugglerFile);
-
             result.Files.CurrentFileName = smugglerFile; 
             result.Files.CurrentFile++;
 
             onProgress.Invoke(result.Progress);
-            using (var zip = await GetZipArchiveForSnapshot(lastPath))
+
+            if (_zipArchiveForSnapshot == null)
+                throw new InvalidOperationException($"Restoring of smuggler values failed because {nameof(_zipArchiveForSnapshot)} is null");
+
+            var entry = _zipArchiveForSnapshot.GetEntry(RestoreSettings.SmugglerValuesFileName);
+            if (entry != null)
             {
-                foreach (var entry in zip.Entries)
+                using (_zipArchiveForSnapshot)
+                await using (var input = entry.Open())
+                await using (var inputStream = GetSnapshotInputStream(input, database.Name))
+                await using (var uncompressed = await RavenServerBackupUtils.GetDecompressionStreamAsync(inputStream))
                 {
-                    if (entry.Name == RestoreSettings.SmugglerValuesFileName)
+                    var source = new StreamSource(uncompressed, context, database);
+                    var smuggler = new Smuggler.Documents.DatabaseSmuggler(database, source, destination,
+                        database.Time, smugglerOptions, onProgress: onProgress, token: _operationCancelToken.Token)
                     {
-                        await using (var input = entry.Open())
-                        await using (var inputStream = GetSnapshotInputStream(input, database.Name))
-                        await using (var uncompressed = await RavenServerBackupUtils.GetDecompressionStreamAsync(inputStream))
-                        {
-                            var source = new StreamSource(uncompressed, context, database);
-                            var smuggler = new Smuggler.Documents.DatabaseSmuggler(database, source, destination,
-                                database.Time, smugglerOptions, onProgress: onProgress, token: _operationCancelToken.Token)
-                            {
-                                BackupKind = BackupKind.Incremental
-                            };
+                        BackupKind = BackupKind.Incremental
+                    };
 
-                            await smuggler.ExecuteAsync(ensureStepsProcessed: true, isLastFile: true);
-                        }
-                        break;
-                    }
+                    await smuggler.ExecuteAsync(ensureStepsProcessed: true, isLastFile: true);
                 }
+
+                _zipArchiveForSnapshot = null; // the zip archive is not needed anymore
```

Comment:
this is not in a finally, so you can end up with double disposal in case of exception, is that ok?